### PR TITLE
SEARCH-334 searchField support and SEARCH-376

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose

Add searchField attribute to mark field as accepting array/values mix as for arangosearch links.
Fixes assertion for array comparison handling - as aql allows to have not only array but any expression as value argument, wrong assertion removed, tests added.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [x] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: No enterprise part
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/IResearch/Search.cpp [hunk 1]
```diff
@@ -196,6 +196,15 @@ std::string abstractCheckFields(auto const& lhs, auto const& rhs,
       continue;
     }
     TRI_ASSERT(it->first == prefix);
+
+    if (SkipSameAnalyzer &&
+        it->second.isSearchField != field.second.isSearchField) {
```
Comment (MBkkt) [inline arangod/IResearch/Search.cpp:201]:
```suggestion
        (it->second.isSearchField != field.second.isSearchField && (prefix == name || prefix,includeAllFields))) {
```

File: arangod/IResearch/Search.cpp [hunk 2]
```diff
@@ -196,6 +196,15 @@ std::string abstractCheckFields(auto const& lhs, auto const& rhs,
...
+      return absl::StrCat(rhsIs, " field '", name, "' searchField '",
+                          field.second.isSearchField, "' mismatches ", lhsIs,
+                          " field searchField '", it->second.isSearchField,
+                          "'");
+    }
+
```
Comment (MBkkt) [inline arangod/IResearch/Search.cpp:207]: auto sameAnalyzer, sameSearchField = ...

File: arangod/IResearch/Search.cpp [hunk 3]
```diff
@@ -196,6 +196,15 @@ std::string abstractCheckFields(auto const& lhs, auto const& rhs,
       continue;
     }
```
Comment (MBkkt) [inline arangod/IResearch/Search.cpp:197]:
```cpp
SameCollection = !SkipSameAnalyzer;
if (l.collection == r.collection || l.search != r.search || l.analyzer != r.analyzer) {
  if (prefix.size() == name.size()) {
   if (SameCollection) {
     return error1;
   } else if (l.search != r.search) {
     return error2;
   } else {
     return error3;
   }
  } else if (prefix.includeAllFields) {
   return error;
  }
}
```

**Review by MBkkt — CHANGES_REQUESTED** [review]

**Review by gnusi — APPROVED** [review]

**Comment by Dronplane** [comment]
tests blue except know failure
 * Test "chaos"
    [FAILED]  tests\js\client\chaos\test-module-chaos-load-cluster.js__with_IntermediateCommits_with_FailurePoints_with_Delays_with_StreamingTransactions_with_Truncate_with_VaryingOverwriteMode

      "testWorkInParallel_with_IntermediateCommits_with_FailurePoints_with_Delays_with_StreamingTransactions_with_Truncate_with_VaryingOverwriteMode" failed: Error: at assertion #603: Shards were not getting in sync in time, giving up!: (false) does not evaluate to true

**Pull request merged**